protocol/v2/ssv/runner/proposer.go

```diff
@@ -105,33 +104,11 @@ func (r *ProposerRunner) ProcessPreConsensus(logger *zap.Logger, signedMsg *spec
 	logger.Debug("🧩 reconstructed partial RANDAO signatures",
 		zap.Uint64s("signers", getPreConsensusSigners(r.GetState(), root)))
 
-	var ver spec.DataVersion
-	var obj ssz.Marshaler
 	var start = time.Now()
-	if r.ProducesBlindedBlocks {
-		// get block data
-		obj, ver, err = r.GetBeaconNode().GetBlindedBeaconBlock(duty.Slot, r.GetShare().Graffiti, fullSig)
-		if err != nil {
-			// Prysm workaround: when Prysm can't retrieve an MEV block, it responds with an error
-			// saying the block isn't blinded, implying to request a standard block instead.
-			// https://github.com/prysmaticlabs/prysm/issues/12103
-			if nodeClientProvider, ok := r.GetBeaconNode().(goclient.NodeClientProvider); ok &&
-				nodeClientProvider.NodeClient() == goclient.NodePrysm {
-				logger.Debug("failed to get blinded beacon block, falling back to standard block")
-				obj, ver, err = r.GetBeaconNode().GetBeaconBlock(duty.Slot, r.GetShare().Graffiti, fullSig)
-				if err != nil {
-					return errors.Wrap(err, "failed falling back from blinded to standard beacon block")
-				}
-			} else {
-				return errors.Wrap(err, "failed to get blinded beacon block")
-			}
-		}
-	} else {
-		// get block data
-		obj, ver, err = r.GetBeaconNode().GetBeaconBlock(duty.Slot, r.GetShare().Graffiti, fullSig)
-		if err != nil {
-			return errors.Wrap(err, "failed to get beacon block")
-		}
+	// get block data
+	obj, ver, err := r.fetchBeaconBlock(logger, duty.Slot, fullSig)
+	if err != nil {
+		return errors.Wrap(err, "failed to get beacon block")
```

Comment:
It's part of old code, but I'd like to suggest using `fmt.Errorf` for new code/changes as [github.com/pkg/errors]( https://github.com/pkg/errors) is discontinued